types/protobuf.go

```diff
@@ -140,14 +139,14 @@ func (tm2pb) Evidence(ev Evidence, valSet *ValidatorSet, evTime time.Time) abci.
 	case *PotentialAmnesiaEvidence:
 		evType = ABCIEvidenceTypePotentialAmnesia
 	default:
-		panic(fmt.Sprintf("Unknown evidence type: %v %v", ev, reflect.TypeOf(ev)))
+		panic(fmt.Sprintf("unknown evidence type: %v %v", ev, reflect.TypeOf(ev)))
 	}
 
 	return abci.Evidence{
 		Type:             evType,
 		Validator:        TM2PB.Validator(val),
 		Height:           ev.Height(),
-		Time:             evTime,
+		Time:             ev.Time(),
```

Comment:
I believe we have to verify time now (since we allow arbitrary values) - "it doesn't solve the problem if we need to verify that time is actually correct, where we'd need to load the old block anyways. So one way or another we need to get access to the old time." in state#VerifyEvidence

Reply:
hmm are you suggesting that we need to load the block for the height that the infraction occurred and use that time? What about all the other times we use ev.Time() for checking expiration?

Reply:
> hmm are you suggesting that we need to load the block for the height that the infraction occurred and use that time?

yep

> What about all the other times we use ev.Time() for checking expiration?

can u point me to them?

Reply:
https://github.com/tendermint/tendermint/blob/3c21c3546c8e1cad89ed7b47f85b842dd98f4a51/state/validation.go#L199

Reply:
https://github.com/tendermint/tendermint/blob/3c21c3546c8e1cad89ed7b47f85b842dd98f4a51/evidence/pool.go#L287

Reply:
> yep

That's annoying

Reply:
make sure to read https://github.com/tendermint/tendermint/issues/4150 for more context.

Reply:
@melekes, we've decided to separate the issue such that this PR just addresses the issue of the wrong usage of time (using time it was committed on the block instead of the time of infraction). In another PR we will look to change where evidence sources it's time (i.e. currently with votes) to a more reliable source.